src/com/goide/util/GoUtil.java

```diff
@@ -195,9 +195,9 @@ public static boolean couldBeReferenceTo(@NotNull PsiElement definition, @NotNul
       public Result<Collection<String>> compute() {
         Collection<String> set = ContainerUtil.newLinkedHashSet();
         for (PsiFile file : dir.getFiles()) {
-          if (file instanceof GoFile) {
+          if (allowed(file)) {
```

Comment:
Is this really needed? If it so than `dir` is wrong dependency for cache. Result of `allowed` invocation might change on changing IDE settings